This workbook contains analysis for only those players who have been drafted by NHL Teams

In [2]:
# Dependencies
import pandas as pd
import plotly.express as px

In [3]:
# Read CSV into DF
df =pd.read_csv('data_sets/drafted_data.csv')
df.head()

LEAGUE                  PLAYER_ID POSITION  SEASON  AGE_SEPT_15  GP   G   A  \
0    WHL  100577/richard-nejezchleb        F    2015    21.369444  49  20  31   
1  QMJHL        100867/jan-kostalek        D    2015    20.577778  57   7  36   
2    OHL     101221/dominik-kubalik        F    2014    19.066667  59  18  11   
3    WHL          10123/denis-rehak        D    2004    19.336111  25   0   3   
4   USHL       101430/nathan-walker        F    2013    19.605556  29   7  20   

   TP       PPG  ...  SEASON_>20GP  CUM_GP  ONLY_1_SEASON  MIN_3_SEASONS  \
0  51  1.040816  ...             3     150              0              1   
1  43  0.754386  ...             3     160              0              1   
2  29  0.491525  ...             2     126              0              0   
3   3  0.120000  ...             1      25              1              0   
4  27  0.931034  ...             1      29              1              0   

   DY_STATUS SHOOTS DRAFTED   NHL_PPG  NHL_GP  NHL_DV  
0       DY+3      L     Yes  0.000000       0       0  
1       DY+2      R     Yes  0.000000       0       0  
2       DY+1      L     Yes  0.000000       0       0  
3       DY+1      L     Yes  0.000000       0       0  
4       DY+1      L     Yes  0.166667      12       1  

[5 rows x 21 columns]

In [13]:
# Top 25 players by NHL Games Played, these should be familiar to hockey fans
nhl_df=df[['PLAYER_ID','NHL_PPG','NHL_GP','GP','G','A','TP']].copy()
# Drop players with 0 NHL games
nhl_df= nhl_df[nhl_df.NHL_GP !=0] 
nhl_df.sort_values('NHL_GP', ascending=False).head(25)

PLAYER_ID   NHL_PPG  NHL_GP  GP   G    A   TP
2401    8764/scott-hartnell  0.566053    1249  62  27   55   82
1963   4506/justin-williams  0.631833    1244  68  37   46   83
1807   3659/jay-bouwmeester  0.350507    1184  61  11   50   61
1804        3656/eric-staal  0.829642    1174  66  39   59   98
2038     4723/brad-richards  0.827709    1126  63  71  115  186
2522      9099/dustin-brown  0.553268    1117  58  34   42   76
2507       9040/dan-hamhuis  0.319853    1088  59  10   50   60
2443    8879/brent-seabrook  0.425139    1082  63  12   42   54
2483      9012/duncan-keith  0.541318    1077  37  11   35   46
2395      8715/mike-ribeiro  0.738361    1074  23  18   31   49
2365      8597/nick-schultz  0.164165    1066  59  17   30   47
1808      3661/jason-spezza  0.859155    1065  53  42   63  105
2413  8809/jason-pominville  0.685849    1060  66  57   64  121
1806         3658/rick-nash  0.759434    1060  54  32   40   72
2376      8626/radim-vrbata  0.589404    1057  55  56   64  120
2356      8564/dion-phaneuf  0.471374    1048  55  24   32   56
2426  8840/antoine-vermette  0.492352    1046   4   0    2    2
2523       9103/brent-burns  0.622244    1043  68  15   25   40
2343      8493/thomas-vanek  0.766764    1029  53  46   45   91
2513  9079/patrice-bergeron  0.790856    1028  70  23   50   73
2534      9128/trevor-daley  0.297537    1015  57  20   33   53
2445       8889/matt-stajan  0.411765    1003  57  34   60   94
2485       9014/ryan-kesler  0.572428    1001  13   5    5   10
2082    5528/manny-malhotra  0.297679     991   5   2    2    4
2353       8541/corey-perry  0.785425     988  60  47   83  130

In [14]:
# Top NHL players by NHL Points per game
nhl_df.sort_values('NHL_PPG', ascending=False).head(25)
# These are mostly pretty predictable, with a few outliers having only played a game or two
# Ryan Poehling, the top player by NHL PPG scored a hat trick in his career debut,
# which happened to be the final game of the 2018-2019 season
# which also happens to be the last game recorded in our dataset.

PLAYER_ID   NHL_PPG  NHL_GP  GP   G    A   TP
1593    288045/ryan-poehling  3.000000       1   9   2    2    4
990    183442/connor-mcdavid  1.296167     287  47  44   76  120
2156      6146/sidney-crosby  1.289502     943  62  66  102  168
2607       9326/patrick-kane  1.038760     903  58  62   83  145
2242   77237/nikita-kucherov  1.033557     447  33  29   34   63
168     11113/steven-stamkos  1.026810     746  61  58   47  105
2439   88699/tyler-lewington  1.000000       2  69   9   36   45
2781       9654/ivan-baranka  1.000000       1  64   7   16   23
744       14998/damian-surma  1.000000       2  55  28   27   55
2666   94254/patrick-sieloff  1.000000       2  45   3    8   11
236         11431/carl-sneep  1.000000       1  13   1    3    4
2425   88391/johnny-gaudreau  0.982234     394  60  36   36   72
1143  199898/auston-matthews  0.966981     212  24  20   28   48
2572       9223/john-tavares  0.944075     751  56  58   46  104
2348       8526/ryan-getzlaf  0.938008     984  51  29   25   54
1270  223194/mitchell-marner  0.929461     241  57  39   77  116
71       10437/claude-giroux  0.928136     821  55  38   68  106
681        14360/taylor-hall  0.909253     562  57  40   66  106
1081      191959/jack-eichel  0.905594     286  24  20   25   45
2209    71913/leon-draisaitl  0.888889     351  32  19   34   53
1010    186310/mathew-barzal  0.885542     166  41  10   69   79
2809  99204/nathan-mackinnon  0.881579     456  44  32   43   75
814         15399/jamie-benn  0.871141     745  56  46   36   82
1349      23807/tyler-seguin  0.870536     672  63  48   58  106
2158     62204/brayden-point  0.864629     229  48  35   53   88

In [6]:
# create a df with only player stats
stats_df = df.drop(['LEAGUE','PLAYER_ID','POSITION','SEASON','AGE_SEPT_15','SEASON_NO','ONLY_1_SEASON','MIN_3_SEASONS','DY_STATUS','SHOOTS','DRAFTED','NHL_DV'], axis=1)
stats_df

GP   G   A   TP       PPG  SEASON_>20GP  CUM_GP   NHL_PPG  NHL_GP
0     49  20  31   51  1.040816             3     150  0.000000       0
1     57   7  36   43  0.754386             3     160  0.000000       0
2     59  18  11   29  0.491525             2     126  0.000000       0
3     25   0   3    3  0.120000             1      25  0.000000       0
4     29   7  20   27  0.931034             1      29  0.166667      12
5     68  14  54   68  1.000000             1      68  0.223529     850
6     68  28  40   68  1.000000             2     134  0.571109     893
7     59  12  37   49  0.830508             4     249  0.000000       2
8     43  13  35   48  1.116279             4     253  0.430052     193
9     56  20  33   53  0.946429             5     312  0.000000       0
10    59   3  21   24  0.406780             2     133  0.000000       0
11    61  24  33   57  0.934426             1      61  0.158730     378
12    18   3   7   10  0.555556             0      40  0.286689     293
13    56  27  28   55  0.982143             4     196  0.185185      27
14    63  13  17   30  0.476190             3     183  0.093220     118
15    68  41  56   97  1.426471             3     208  0.308411     749
16    56  30  47   77  1.375000             2     113  0.787185     874
17    66  16  57   73  1.106061             3     177  0.322280     965
18    51  14  32   46  0.901961             2     115  0.043478      46
19    64  31  41   72  1.125000             3     189  0.237179     156
20    23   4   6   10  0.434783             3     160  0.208134     418
21    62  16  26   42  0.677419             3     190  0.171322     537
22    57  10  51   61  1.070175             4     250  0.213058     291
23    27  23  15   38  1.407407             3     164  0.317726     299
24    64   6  25   31  0.484375             3     200  0.000000       0
25    66  25  59   84  1.272727             1      66  0.582139     907
26    55  47  33   80  1.454545             1      55  0.435897     156
27    58  19  15   34  0.586207             4     253  0.202296     697
28    60  35  33   68  1.133333             3     162  0.111111       9
29    66  29  64   93  1.409091             4     212  0.000000       0
...   ..  ..  ..  ...       ...           ...     ...       ...     ...
2780  59  40  91  131  2.220339             2     122  0.498584     353
2781  64   7  16   23  0.359375             2     122  1.000000       1
2782  69  22  30   52  0.753623             1      69  0.294118      17
2783  55  33  16   49  0.890909             1      55  0.000000       0
2784  50   5   6   11  0.220000             1      50  0.000000       0
2785   4   1   2    3  0.750000             0      61  0.000000       0
2786  65  43  34   77  1.184615             3     198  0.250000       8
2787  25   0   5    5  0.200000             2      61  0.184211     114
2788  27   7  14   21  0.777778             4     162  0.032787      61
2789  72  67  91  158  2.194444             4     250  0.441176      68
2790  59   8  21   29  0.491525             4     206  0.000000       0
2791  68  45  83  128  1.882353             3     199  0.391304     253
2792  36  10  23   33  0.916667             4     221  0.315412     279
2793  35   0  10   10  0.285714             4     202  0.076923     130
2794  26   3  10   13  0.500000             2      59  0.465409     159
2795  62   5  28   33  0.532258             2     139  0.372168     309
2796  59  26  24   50  0.847458             4     229  0.000000       0
2797  58  17  43   60  1.034483             4     256  0.217822     101
2798  61  12  47   59  0.967213             2     123  0.181102     127
2799  63  25  27   52  0.825397             4     253  0.000000       0
2800  17   6  15   21  1.235294             0      77  0.386243     189
2801  59  21  27   48  0.813559             3     160  0.378947     285
2802  48  23  35   58  1.208333             3     125  0.317181     454
2803  26  15  19   34  1.307692             4 

In [17]:
# Mean statistics for all drafted players
stats_df.mean(axis=0)

GP               54.110676
G                18.587544
A                29.658363
TP               48.245196
PPG               0.871936
SEASON_>20GP      2.888968
CUM_GP          171.701423
NHL_PPG           0.118336
NHL_GP          105.266548
dtype: float64

In [30]:
# Median statistics of a player who has been drafted by an NHL Team, to try and account for outliers like Poehling
stats_median = stats_df.median(axis=0)
stats_median

GP               58.000000
G                16.000000
A                28.000000
TP               46.000000
PPG               0.842105
SEASON_>20GP      3.000000
CUM_GP          185.000000
NHL_PPG           0.000000
NHL_GP            0.000000
dtype: float64

In [39]:
# Determine what pct of drafted players make the nhl
# Group by NHL/ non-NHL
drafted_vs_nhl_df = df.groupby('NHL_DV')
# Count all Players
player_ct = df['PLAYER_ID'].count()
# Count each group (nhl/non-nhl)
split_player_ct = drafted_vs_nhl_df['PLAYER_ID'].count()
# Count NHL players
nhl_ct = drafted_vs_nhl_df['PLAYER_ID'].count()
# Calculate pct
nhl_pct = (nhl_ct/player_ct)
# Create new DF for counts
nhl_pct_df = pd.DataFrame({'Player Count': split_player_ct,
                         'Percent of Playerbase': nhl_pct})

# Format percents with %
nhl_pct_df['Percent of Playerbase'] = nhl_pct_df['Percent of Playerbase'].astype(float).map("{:.2%}".format)

nhl_pct_df.transpose()

NHL_DV                      0       1
Player Count             1606    1204
Percent of Playerbase  57.15%  42.85%

Only 42.85% of players who were drafted by NHL teams ever make it into the NHL

In [48]:
# Comparing average juniors statistics for a player who does/ does not make it to the NHL level & percent who make it

# GP AVG
gp_avg= drafted_vs_nhl_df['GP'].mean()
# Goals AVG
g_avg= drafted_vs_nhl_df['G'].mean()
# Assits AVG
a_avg= drafted_vs_nhl_df['A'].mean()
# Total Points
pt_avg= drafted_vs_nhl_df['TP'].mean()
# Points per Game avg
ppg_avg= drafted_vs_nhl_df['PPG'].mean()


compared_stats_df = pd.DataFrame({'Player Count': split_player_ct,
                         'Percent of Playerbase': nhl_pct,
                            'Average Juniors Games Played': gp_avg,
                            'Average Juniors Goals': g_avg,
                            'Average Juniors Assists': a_avg,
                            'Average Juniors Points': pt_avg,
                            'Average Juniors Points Per Game': ppg_avg})

# Formatting
compared_stats_df['Percent of Playerbase'] = compared_stats_df['Percent of Playerbase'].astype(float).map("{:.2%}".format)
pd.options.display.float_format = '{:,.2f}'.format
compared_stats_df.transpose()

NHL_DV                                0       1
Player Count                       1606    1204
Percent of Playerbase            57.15%  42.85%
Average Juniors Games Played      54.22   53.96
Average Juniors Goals             15.82   22.28
Average Juniors Assists           25.74   34.88
Average Juniors Points            41.56   57.16
Average Juniors Points Per Game    0.74    1.05

In [ ]:
# Graph junior PPG vs NHL GP